plugins/ingest/src/test/java/org/elasticsearch/plugin/ingest/PipelineStoreTests.java

```diff
@@ -173,4 +184,49 @@ public int compare(PipelineStore.PipelineReference first, PipelineStore.Pipeline
         assertThat(result.get(1).getPipeline().getId(), equalTo("bar"));
     }
 
+    ActionFuture<SearchResponse> expectedSearchReponse(List<SearchHit> hits) {
+        return new PlainActionFuture<SearchResponse>() {
+
+            @Override
+            public SearchResponse get(long timeout, TimeUnit unit) {
+                InternalSearchHits hits1 = new InternalSearchHits(hits.toArray(new InternalSearchHit[0]), hits.size(), 1f);
+                return new SearchResponse(new InternalSearchResponse(hits1, null, null, false, null), "_scrollId", 1, 1, 1, null);
+            }
+        };
+    }
+
+    ActionFuture<GetResponse> expectedGetResponse(boolean exists) {
+        return new PlainActionFuture<GetResponse>() {
+            @Override
+            public GetResponse get() throws InterruptedException, ExecutionException {
+                return new GetResponse(new GetResult("_index", "_type", "_id", 1, exists, null, null));
+            }
+        };
+    }
+
+    GetRequest eqGetRequest(String index, String type, String id) {
+        return Matchers.argThat(new GetRequestMatcher(index, type, id));
+    }
+
+    static class GetRequestMatcher extends ArgumentMatcher<GetRequest> {
```

Comment:
I can make it an anonymous inner class, but then the method gets less clean IMO, so I prefer it to be a separate inner class.